packages/api-v2/src/APIProvider.tsx

```diff
@@ -218,6 +222,15 @@ const APIProvider = ({ children }: PropsWithChildren<TAPIProviderProps>) => {
         return () => clearTimeout(reconnectTimerId);
     }, [reconnect]);
 
+    // reconnects to latest WS url for new language only when language changes
+    useEffect(() => {
+        if (prevLanguage !== language) {
+            setReconnect(true);
+            setPrevLanguage(language);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [language]);
```

Comment:
Added logic in APIProvider to reconnect to new WS url when language is switched so that API values are returned in the correct language (i.e transfer error messages below):

https://github.com/user-attachments/assets/1bbc03e2-b100-4837-aa28-7acd65124216